src/osd/scrubber/ScrubStore.h

```diff
@@ -45,18 +44,23 @@ class Store {
     uint64_t max_return) const;
 
  private:
-  Store(const coll_t& coll, const ghobject_t& oid, ObjectStore* store);
+  Store(const coll_t& coll, const ghobject_t& oid, const ghobject_t& deep_oid, ObjectStore* store);
   std::vector<ceph::buffer::list> get_errors(const std::string& start,
 					     const std::string& end,
 					     uint64_t max_return) const;
  private:
   const coll_t coll;
   const ghobject_t hoid;
+  const ghobject_t deep_hoid;
   // a temp object holding mappings from seq-id to inconsistencies found in
   // scrubbing
   OSDriver driver;
+  OSDriver deep_driver;
   mutable MapCacher::MapCacher<std::string, ceph::buffer::list> backend;
+  mutable MapCacher::MapCacher<std::string, ceph::buffer::list> deep_backend;
   std::map<std::string, ceph::buffer::list> results;
+  std::map<std::string, ceph::buffer::list> deep_results;
+
```

Comment:
I would love to get this documented (`results` stores all, `deep_results` only `has_deep_errors() == true`). Actually, this is a good place to describe the design as it has been in our discussions.